src/mediapipe_internal/mediapipegraphconfig.hpp

```diff
@@ -108,20 +116,25 @@ class MediapipeGraphConfig {
          * @param graphPath
          */
     void setGraphPath(const std::string& graphPath) {
-        this->graphPath = graphPath;
-
-        if (!FileSystem::isLocalFilesystem(graphPath)) {
-            // Cloud filesystem
-            this->graphPath = graphPath;
-        } else if (graphPath.at(0) == '/') {
-            // Full path case
-            this->graphPath = graphPath;
-        } else {
-            // Relative path case
-            if (this->rootDirectoryPath.empty())
-                throw std::logic_error("Using graph relative path without setting configuration directory path.");
-            this->graphPath = this->rootDirectoryPath + graphPath;
-        }
+        FileSystem::setPath(this->graphPath, graphPath, this->rootDirectoryPath);
```

Comment:
Move implementation to cpp file and then we don't need to include filesystem here.